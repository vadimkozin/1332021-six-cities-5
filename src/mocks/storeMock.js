import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
