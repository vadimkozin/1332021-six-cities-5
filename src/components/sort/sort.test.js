import React from 'react';
import {Provider} from 'react-redux';
import {render} from 'enzyme';
import storeMock from '../../mocks/storeMock';
import Sort from './sort';

describe(`Sort`, () => {
  it(`should render Sort component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <Sort />
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
