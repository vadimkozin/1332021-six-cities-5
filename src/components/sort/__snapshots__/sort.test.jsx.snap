// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort should render Sort component 1`] = `
<form
  action="#"
  class="places__sorting"
  method="get"
>
  <span
    class="places__sorting-caption"
  >
    Sort by Â 
  </span>
  <span
    class="places__sorting-type"
    tabindex="0"
  >
    Popular
    <svg
      class="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        href="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    class="places__options places__options--custom places__options--opened"
  >
    <li
      class="places__option places__option--active"
      data-sort="popular"
      tabindex="0"
    >
      Popular
    </li>
    <li
      class="places__option false"
      data-sort="to-high"
      tabindex="0"
    >
      Price: low to high
    </li>
    <li
      class="places__option false"
      data-sort="to-low"
      tabindex="0"
    >
      Price: high to low
    </li>
    <li
      class="places__option false"
      data-sort="top-rated"
      tabindex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
